lib/src/auth/mod.rs

```diff
@@ -0,0 +1,12 @@
+mod static_client_certificate_provider;
+
+#[derive(Debug, Clone)]
+pub struct ClientCertificate {
+    pub(crate) cert_file: String,  // Path to the TLS certificate file.
+}
+
+pub trait ClientCertificateProvider {
```

Comment:
I'm not sure what the benefit of the trait is. There is only one implementation and since the constructor has a `pub(crate)` fields, users cannot implement their own since they cannot create a `ClientCertificate`. And if that field is pub, it all seems like a complicated way to provide a path.

Instead, I would remove the Provider trait and the static impl, keep the ClientCertificate struct and add a `pub new(path: impl AsRef<Path>)` to it, maybe also with an according From impl.

Reply:
Yeah, the idea was to have different providers for certificates, but I understand is a bit too much at the moment. I'll remove this